include/openmc/particle.h

```diff
@@ -122,6 +124,9 @@ struct MacroXS {
   double absorption;    //!< macroscopic absorption xs
   double fission;       //!< macroscopic fission xs
   double nu_fission;    //!< macroscopic production xs
+  double prompt_nu_fission;//!< macroscopic prompt production
+  std::vector<double> delayed_nu_fission;//!< macroscopic delayed production
```

Comment:
Why isn't this `std::array` like in the micro?

Reply:
When I use p.macro_xs_.delayed_nu_fission later in src/mgxs.cpp I have to resize the vector to exactly the number of actual delayed groups to avoid errors. An array, being fixed size, doesn't have that capability from what I read. 
When I submit those changes to mgxs.cpp in a future PR, if you see a workaround, I am happy to modify it from a vector to an array. But for now, I don't see a way to make it an array.

Reply:
That makes sense -- we can leave it as `vector`